tests/unit/models/per-bundle-addon-cache/proxy-test.js

```diff
@@ -271,29 +271,317 @@ describe('models/per-bundle-addon-cache', function () {
       expect(byName['test-addon-b'].realAddonInstanceCount).to.equal(1);
       expect(byName['test-addon-b'].proxyCount).to.equal(3);
 
-      let cacheEntries = findAddonCacheEntriesByName(project.perBundleAddonCache, 'lazy-engine-a', 'test-addon-a');
-      expect(cacheEntries).to.exist;
+      const lazyEngineAPkgInfo = project.addons.find((addon) => addon.name === 'lazy-engine-a')._packageInfo;
+      let cacheEntries = findAddonCacheEntriesByName(project.perBundleAddonCache, lazyEngineAPkgInfo, 'test-addon-a');
+      expect(cacheEntries).to.deep.equal([], 'should exist; lazy-engine A has opted-in to bundle caching');
 
-      // project cache should be used
-      expect(cacheEntries.length).to.equal(0);
+      cacheEntries = findAddonCacheEntriesByName(project.perBundleAddonCache, lazyEngineAPkgInfo, 'test-addon-b');
+      expect(cacheEntries).to.deep.equal([], 'should exist; lazy-engine A has opted-in to bundle caching');
 
-      cacheEntries = findAddonCacheEntriesByName(project.perBundleAddonCache, 'lazy-engine-a', 'test-addon-b');
-      expect(cacheEntries).to.exist;
+      const lazyEngineBPkgInfo = project.addons.find((addon) => addon.name === 'lazy-engine-b')._packageInfo;
+      cacheEntries = findAddonCacheEntriesByName(project.perBundleAddonCache, lazyEngineBPkgInfo, 'test-addon-a');
+      expect(cacheEntries).to.deep.equal([], 'should exist; lazy engine B has opted into bundle caching');
 
-      // project cache should be used
-      expect(cacheEntries.length).to.equal(0);
+      cacheEntries = findAddonCacheEntriesByName(project.perBundleAddonCache, lazyEngineBPkgInfo, 'test-addon-b');
+      expect(cacheEntries).to.deep.equal([], 'should exist; lazy engine B has opted into bundle caching');
+      expect(cacheEntries.length).to.equal(0, 'project cache should be used');
```

Comment:
Is this necessary, since you have the 'to.deep.equal' directly above?  Seems superfluous.

Reply:
Also, should there be positive tests here? The above shows that the 2 `test-addon...` objects aren't in the lazy engine caches - should it be shown that they ARE in the project cache?